tgui/packages/tgui/interfaces/ShuttleBlueprints.tsx

```diff
@@ -0,0 +1,458 @@
+import { BooleanLike } from 'common/react';
+import { ReactNode, useState } from 'react';
+
+import { useBackend } from '../backend';
+import {
+  Box,
+  Button,
+  Dropdown,
+  Input,
+  Section,
+  Stack,
+  Tooltip,
+} from '../components';
+import { Direction } from '../constants';
+import { Window } from '../layouts';
+
+type AreaData = { name: string; ref: string };
+
+type VisualizationToggleProps = { visualizing: BooleanLike };
+
+type ShuttleConstructionProps = Partial<{
+  visualizing: BooleanLike;
+  onShuttleFrame: BooleanLike;
+  tooManyShuttles: BooleanLike;
+  onCustomShuttle: BooleanLike;
+  masterExists: BooleanLike;
+}>;
+
+type ShuttleConfigurationProps = Partial<{
+  visualizing: BooleanLike;
+  onShuttleFrame: BooleanLike;
+  onShuttle: BooleanLike;
+  inDefaultArea: BooleanLike;
+  currentArea: AreaData;
+  defaultApc: BooleanLike;
+  apcInMergeRegion: BooleanLike;
+  apcs: Map<string, BooleanLike>;
+  neighboringAreas: Map<string, string>;
+  idle: BooleanLike;
+  masterExists: BooleanLike;
+}>;
+
+type ShuttleBlueprintsData = Partial<{
+  linkedShuttle: string;
+  shuttles: Map<string, string>;
+}> &
+  ShuttleConstructionProps &
+  ShuttleConfigurationProps;
```

Comment:
BYOND data should never be all partial.
Yes you are sending data which is conditional but this would be much cleaner as a discriminated union